mlir-sycl/lib/Dialect/IR/SYCLOps.cpp

```diff
@@ -89,6 +89,21 @@ bool SYCLAddrSpaceCastOp::areCastCompatible(TypeRange inputs,
           (outputMS == genericAddressSpace));
 }
 
+LogicalResult SYCLAccessorGetPointerOp::verify() {
+  const auto accTy = getOperand()
+                         .getType()
+                         .cast<MemRefType>()
+                         .getElementType()
+                         .cast<AccessorType>();
+  const Type resTy = getResult().getType();
+  const Type resElemTy = resTy.cast<MemRefType>().getElementType();
+  return (resElemTy != accTy.getType())
+             ? emitOpError(
+                   "Expecting a reference to this accessor's value type (")
+                   << accTy.getType() << "). Got " << resTy
+             : success();
```

Comment:
Just two minors:
1. Can we also check the memory spaces are the same? (also add checks in invalid.mlir for this)
2. Can you use `cast<X>(Y)` instead of `Y.cast<X>()` (the latter [is deprecated](https://mlir.llvm.org/deprecation/))

Reply:
LGTM after these two are fixed

Reply:
I thought we discussed that we don't want to do the first one?